File: Leetcode - Java/643_array_findMaxAverage.java
// https://leetcode.com/problems/maximum-average-subarray-i/

class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int i = 0, windowSum = 0;
        double maxAve = -Double.MAX_VALUE; 
        
        for (int j = 0; j < nums.length; j++) {
            windowSum += nums[j];
            if (j >= k - 1) {
                double avg = windowSum / 1.0 / k;
                maxAve = Math.max(maxAve, avg);
                windowSum -= nums[i++];
            }
        }
        return maxAve;
    }
}

File: Leetcode-Python-New/643_array_findMaxAverage.py
# https://leetcode.com/problems/maximum-average-subarray-i/

class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        max_avg = -float('inf')
        i, windowSum = 0, 0.0
        
        for j in range(len(nums)):
            windowSum += nums[j]
            if j >= k - 1:
                avg = windowSum / k
                max_avg = max(max_avg, avg)
                windowSum -= nums[i]
                i += 1
                
        return max_avg